insights/client/config.py

```diff
@@ -681,6 +682,52 @@ def _validate_options(self):
         '''
         Make sure there are no conflicting or invalid options
         '''
+
+        def _validate_obfuscation_options():
+            # When old switches are set explicitly, even set as False
+            if self.obfuscate is not None or self.obfuscate_hostname is not None:
+                # Warning deprecation only on RHEL 8+ and from egg v 3.6.0 (planned)
+                if self._print_errors and get_rhel_version() > 7:
+                    egg_ver = get_version_info()['core_version'].split('-')[0]
+                    if [int(i) for i in egg_ver.split('.')] >= [3, 6, 0]:
+                        sys.stdout.write(
+                            'WARNING: `obfuscate` and `obfuscate_hostname` are deprecated, please use `obfuscation_list` instead.\n'
+                        )
```

Comment:
I would rather to use here `logging.warning()` instead of `sys.stdout.write("WARNING: ...")`. Then we can see the logs in structured way in a log file.

Reply:
Yup, using `sys.stdout.write` was just to keep the uniform style in this `insights.client.config` python module, as it was used in other places as well. If you are good with that, I'm happy to change all the new addeds and the existing others to `logger.warning`.

Reply:
I understand that it may not be consistent with the style in this file, but IMO we are trying to have `logging` module instead of just printing it to console by using `sys.stdout`.